airbyte-cdk/python/airbyte_cdk/test/utils/data.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from pathlib import Path as FilePath
+
+
+def _get_unit_tests_root_folder(execution_folder: FilePath) -> FilePath:
```

Comment:
This is duplication from `_get_unit_test_folder` in `response_builder.py`. 

`read_json_contents` is also duplication for `find_template` in `response_builder.py`. Why not re-use these?

Reply:
I renamed the reader for better semantics. I created a new version of these functions because I believe they should be in a more generic location. Also the reader is slightly different, reading as string, and it can read files with any extension (thinking of plain text, or csv, etc) to make it a bit more flexible.

Reply:
Following this comment, my understanding is:
* _get_unit_tests_root_folder should be common to avoid duplication
* the description of `read_resource_file_contents` should not mention `json`

I'm fine with having `read_resource_file_contents` independent of `find_template` though

Reply:
Thanks for the catches. I removed the wrong referencing of JSON. However, since the _get_unit_tests is private to the RequestBuilder module, I wouldn't want to make it public and refactor. I made the one in this module public, for others to reuse.

Reply:
> I wouldn't want to make it public and refactor

Why not? What are the consequences of doing so?

Reply:
I didn't want to affect the scope design, but you're right. There's not a lot of uses so, did the refactor.